src/algebra/star/subalgebra.lean

```diff
@@ -41,8 +43,162 @@ instance : has_top (star_subalgebra R A) :=
 
 instance : inhabited (star_subalgebra R A) := ⟨⊤⟩
 
+instance : star_mem_class (star_subalgebra R A) A :=
+{ star_mem := λ s a, s.star_mem' }
+
+instance : subsemiring_class (star_subalgebra R A) A :=
+{ add_mem := add_mem',
+  mul_mem := mul_mem',
+  one_mem := one_mem',
+  zero_mem := zero_mem' }
+
+-- this uses the `has_star` instance `s` inherits from `star_mem_class (star_subalgebra R A) A`
+instance (s : star_subalgebra R A) : star_ring s :=
+{ star := star,
+  star_involutive := λ r, subtype.ext (star_star r),
+  star_mul := λ r₁ r₂, subtype.ext (star_mul r₁ r₂),
+  star_add := λ r₁ r₂, subtype.ext (star_add r₁ r₂) }
+
+instance (s : star_subalgebra R A) : algebra R s := s.to_subalgebra.algebra'
+
+instance (s : star_subalgebra R A) : star_module R s :=
+{ star_smul := λ r a, subtype.ext (star_smul r a) }
+
+@[simp]
+lemma mem_carrier {s : star_subalgebra R A} {x : A} : x ∈ s.carrier ↔ x ∈ s := iff.rfl
+
+@[ext] theorem ext {S T : star_subalgebra R A} (h : ∀ x : A, x ∈ S ↔ x ∈ T) : S = T :=
+set_like.ext h
+
+@[simp] lemma mem_to_subalgebra {S : star_subalgebra R A} {x} : x ∈ S.to_subalgebra ↔ x ∈ S :=
+iff.rfl
+
+@[simp] lemma coe_to_subalgebra (S : star_subalgebra R A) : (S.to_subalgebra : set A) = S := rfl
+
+theorem to_subalgebra_injective :
+  function.injective (to_subalgebra : star_subalgebra R A → subalgebra R A) :=
+λ S T h, ext $ λ x, by rw [← mem_to_subalgebra, ← mem_to_subalgebra, h]
+
+theorem to_subalgebra_inj {S U : star_subalgebra R A} : S.to_subalgebra = U.to_subalgebra ↔ S = U :=
+to_subalgebra_injective.eq_iff
+
+/-- Copy of a star subalgebra with a new `carrier` equal to the old one. Useful to fix definitional
+equalities. -/
+protected def copy (S : star_subalgebra R A) (s : set A) (hs : s = ↑S) : star_subalgebra R A :=
+{ carrier := s,
+  add_mem' := λ _ _, hs.symm ▸ S.add_mem',
+  mul_mem' := λ _ _, hs.symm ▸ S.mul_mem',
+  algebra_map_mem' := hs.symm ▸ S.algebra_map_mem',
+  star_mem' := λ _, hs.symm ▸ S.star_mem' }
+
+@[simp] lemma coe_copy (S : star_subalgebra R A) (s : set A) (hs : s = ↑S) :
+  (S.copy s hs : set A) = s := rfl
+
```

Comment:
```suggestion
@[simp] lemma coe_copy (S : star_subalgebra R A) (s : set A) (hs : s = ↑S) :
  (S.copy s hs : set A) = s := rfl

lemma copy_eq (S : star_subalgebra R A) (s : set A) (hs : s = ↑S) : S.copy s hs = S :=
set_like.coe_injective hs

```
Just to match [subalgebra.copy_eq](https://leanprover-community.github.io/mathlib_docs/algebra/algebra/subalgebra/basic.html#subalgebra.copy_eq)